staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -39,21 +39,16 @@ type AuditContext struct {
 	RequestAuditConfig RequestAuditConfig
 
 	// Event is the audit Event object that is being captured to be written in
-	// the API audit log. It is set to nil when the request is not being audited.
-	Event *auditinternal.Event
+	// the API audit log.
+	Event auditinternal.Event
 
-	// annotations holds audit annotations that are recorded before the event has been initialized.
-	// This is represented as a slice rather than a map to preserve order.
-	annotations []annotation
-	// annotationMutex guards annotations AND event.Annotations
+	// annotationMutex guards event.Annotations
 	annotationMutex sync.Mutex
-
-	// auditID is the Audit ID associated with this request.
-	auditID types.UID
 }
 
-type annotation struct {
-	key, value string
+// Enabled checks whether auditing is enabled for this audit context.
+func (ac *AuditContext) Enabled() bool {
+	return ac != nil && ac.Event.Level != auditinternal.LevelNone
```

Comment:
I think the initial default should return `false`, right?
```
func TestEnabled(t *testing.T) {
	ac := &AuditContext{}
	if ac.Enabled() {
		t.Errorf("expected the default to return false")
	}
}
```
maybe add `len(ac.Event.Level) > 0`

Reply:
No, this is intentional. If annotations (or any other request metadata) are recorded before the policy is evaluated, we want to store them. Once the policy is evaluated to `None` we can stop doing the work in that case. I'll add a comment to this affect.